x-pack/plugins/enterprise_search/public/applications/app_search/app_logic.ts

```diff
@@ -14,53 +14,33 @@ import { ConfiguredLimits, Account, Role } from './types';
 import { getRoleAbilities } from './utils/role';
 
 interface AppValues {
-  hasInitialized: boolean;
   ilmEnabled: boolean;
-  configuredLimits: Partial<ConfiguredLimits>;
-  account: Partial<Account>;
-  myRole: Partial<Role>;
+  configuredLimits: ConfiguredLimits;
+  account: Account;
+  myRole: Role;
 }
 interface AppActions {
-  initializeAppData(props: InitialAppData): Required<InitialAppData>;
   setOnboardingComplete(): boolean;
 }
 
-export const AppLogic = kea<MakeLogicType<AppValues, AppActions>>({
+export const AppLogic = kea<MakeLogicType<AppValues, AppActions, Required<InitialAppData>>>({
   path: ['enterprise_search', 'app_search', 'app_logic'],
   actions: {
-    initializeAppData: (props) => props,
     setOnboardingComplete: () => true,
   },
-  reducers: {
-    hasInitialized: [
-      false,
-      {
-        initializeAppData: () => true,
-      },
-    ],
+  reducers: ({ props }) => ({
     account: [
-      {},
+      props.appSearch,
       {
-        initializeAppData: (_, { appSearch: account }) => account || {},
         setOnboardingComplete: (account) => ({
           ...account,
           onboardingComplete: true,
         }),
       },
     ],
-    configuredLimits: [
-      {},
-      {
-        initializeAppData: (_, { configuredLimits }) => configuredLimits?.appSearch || {},
-      },
-    ],
-    ilmEnabled: [
-      false,
-      {
-        initializeAppData: (_, { ilmEnabled }) => !!ilmEnabled,
-      },
-    ],
-  },
+    configuredLimits: [props.configuredLimits.appSearch, {}],
+    ilmEnabled: [props.ilmEnabled, {}],
```

Comment:
(thinking out loud) it seems wrong that these are a part of the logic but never change.

Reply:
I had this hesitation at first too, but I think @JasonStoltz managed to convince me it's not all that weird. We do this in a decent amount of our shared logic files, e.g. Kibana values/helpers, and it's easier to think about it as 'shared context' and not 'shared state' (i.e., it doesn't necessarily need to change or update). Kea at this point is basically a standard way of hooking into / sharing static _or_ dynamic data across all our plugins, and from a dev QOL perspective makes accessing that data quick and easy (e.g., no need to use multiple Provider wrappers).

Reply:
Yeah, this is a bit like you were talking about the other day Byron, about putting configuration in a React Context to pass down throughout your component tree.

This is the same as that, basically instead of using a React Context directly, you're using Kea. And the nice thing about that is that you don't have to deal with Contexts AND Kea in your code, you just always use Kea.